packages/p2p/src/components/modal-manager/modals/email-link-verified-modal/email-link-verified-modal.jsx

```diff
@@ -15,31 +15,45 @@ const EmailLinkVerifiedModal = () => {
     );
 
     return (
-        <Modal has_close_icon is_open={is_modal_open} renderTitle={() => <></>} toggleModal={hideModal} width='440px'>
-            <Modal.Body className='email-verified-modal'>
-                <Icon icon='IcEmailVerificationLinkValid' size='128' />
-                <Text className='email-verified-modal--text' color='prominent' size='s' weight='bold'>
-                    <Localize i18n_default_text="We've verified your order" />
-                </Text>
-                <Text align='center' color='prominent' size='s'>
-                    <Localize
-                        i18n_default_text="Please ensure you've received {{amount}} {{local_currency}} in your account and hit Confirm to complete the transaction."
-                        values={{ amount, local_currency }}
-                    />
-                </Text>
-            </Modal.Body>
-            <Modal.Footer className='email-verified-modal--footer'>
-                <Button
-                    large
-                    primary
-                    onClick={() => {
-                        hideModal({ should_hide_all_modals: true });
-                        order_store.confirmOrder(is_buy_order_for_user);
-                    }}
-                >
-                    <Localize i18n_default_text='Confirm' />
-                </Button>
-            </Modal.Footer>
+        <Modal
```

Comment:
```suggestion
       { order_store.order_information && <Modal
```

Reply:
Will it works like this? so we don't need to show the loader

Reply:
I think for this scenario i need to show the loader here, since order_information can be empty and its better to show that something is loading before showing the modal 😄 

also i wouldve showed the LoadingModal but because of how all the socket calls are, I instead checked it in the modal itself

Reply:
Its actually should do the same 😆 , we wait for order_information to not be empty and show the modal. So, in case websocket call fails to fetch order_information it will not shows the modal loads forever 😅

Reply:
hmmm ok let me see first the flow

Reply:
yeah I'll go with this suggestionnnn i think this should also fix it 👍